src/main/scala/edu/cmu/cs/obsidian/codegen/CodeGenYul.scala

```diff
@@ -283,11 +316,13 @@ object CodeGenYul extends CodeGenerator {
                     s.flatMap(s => {
                         val id_s: Identifier = nextTemp()
                         decl_0exp(id_s) +: translateStatement(s, Some(id_s.name), contractName, checkedTable)
+                        //todo: this also does not assign afterwards; likely the same bug as fixed in IfThenElse
```

Comment:
This sounds serious and potentially hard to debug  later. Is there an issue open for this? If not, please file one and mark it high priority.

Reply:
that's right. i've made an issue for it here and will get to it sooner rather than later. https://github.com/mcoblenz/Obsidian/issues/339